core/src/impl/Kokkos_Atomic_View.hpp

```diff
@@ -339,7 +339,7 @@ class AtomicDataElement {
   }
 
   KOKKOS_INLINE_FUNCTION
-  operator volatile non_const_value_type() volatile const {
```

Comment:
I am not confident that this is ok. The proper solution is to replace it with an atomic_load I guess, in which case you should be able to return non-volatile.

Reply:
@crtrott thanks for the guidance, I'll uncomment line 343 below to use the atomic_load and remove line 344.